src/transport/TcpRemotingClient.cpp

```diff
@@ -1,758 +1,644 @@
-    if (pFuture) {
-      LOG_DEBUG("succ deleted the sync pullrequest for opaque:%d, mq:%s", opaque, mq.toString().data());
-    }
-  } else {
-    LOG_DEBUG("succ deleted the async pullrequest for opaque:%d, mq:%s", opaque, mq.toString().data()); 
-  }
-  //delete the timeout timer for opaque for pullrequest
-  cancelTimerCallback(opaque);
-}
-
-//<!************************************************************************
-}  //<!end namespace;
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "TcpRemotingClient.h"
+#include <stddef.h>
+#if !defined(WIN32) && !defined(__APPLE__)
+#include <sys/prctl.h>
+#endif
+
+#include "Logging.h"
+#include "MemoryOutputStream.h"
+#include "TopAddressing.h"
+#include "UtilAll.h"
+
+namespace rocketmq {
+
+//<!************************************************************************
+TcpRemotingClient::TcpRemotingClient(int pullThreadNum, uint64_t tcpConnectTimeout, uint64_t tcpTransportTryLockTimeout)
+    : m_pullThreadNum(pullThreadNum),
+      m_tcpConnectTimeout(tcpConnectTimeout),
+      m_tcpTransportTryLockTimeout(tcpTransportTryLockTimeout),
+      m_namesrvIndex(0),
+      m_ioServiceWork(m_ioService) {
+
+#if !defined(WIN32) && !defined(__APPLE__)
+  string taskName = UtilAll::getProcessName();
+  prctl(PR_SET_NAME, "NetworkTP", 0, 0, 0);
+#endif
+  for (int i = 0; i != m_pullThreadNum; ++i) {
+    m_threadpool.create_thread(
+        boost::bind(&boost::asio::io_service::run, &m_ioService));
+  }
+#if !defined(WIN32) && !defined(__APPLE__)
+  prctl(PR_SET_NAME, taskName.c_str(), 0, 0, 0);
+#endif
+
+  LOG_INFO("m_tcpConnectTimeout:%ju, m_tcpTransportTryLockTimeout:%ju, m_pullThreadNum:%d",
+           m_tcpConnectTimeout, m_tcpTransportTryLockTimeout, m_pullThreadNum);
+
+  m_async_service_thread.reset(new boost::thread(
+      boost::bind(&TcpRemotingClient::boost_asio_work, this)));
+}
+
+void TcpRemotingClient::boost_asio_work() {
+  LOG_INFO("TcpRemotingClient::boost asio async service running");
+
+#if !defined(WIN32) && !defined(__APPLE__)
+  prctl(PR_SET_NAME, "RemotingAsioT", 0, 0, 0);
+#endif
+
+  // avoid async io service stops after first timer timeout callback
+  boost::asio::io_service::work work(m_async_ioService);
```

Comment:
rename `m_async_ioService` to `m_async_io_service`

Reply:
I don't rename this variable.